source/AI.cpp

```diff
@@ -2854,7 +2854,7 @@ bool AI::DoHarvesting(Ship &ship, Command &command) const
 // Check if this ship should cloak. Returns true if this ship decided to run away while cloaking.
 bool AI::DoCloak(Ship &ship, Command &command)
 {
-	if(ship.Attributes().Get("cloak"))
+	if(ship.CanCloak())
```

Comment:
I would prefer checking the personality first and bailing out instead.

You know what? This whole thing is bad. https://github.com/UnorderedSigh/endless-sky/pull/32

I've used the two things individually instead of calling Ship::CanCloak because I think the decision of whether or not to cloak should be made in AI, not Ship.

Reply:
I removed Ship::CanCloak, and everything checks the personality directly instead.

Reply:
I've merged @warp-core's changes, but they refactored the cloaking logic. I will need to test cloaking to make sure it still works.

Reply:
I did some simple testing, and cloaking appears to be working after @warp-core's changes.